### ISSUE TITLE
[FEATURE] Elvish support for x writer aliases | [mod=elv] [v0.3.14]

### ISSUE THREAD

**Issue by iandol** [x-cmd/x-cmd#31]
**Describe the feature**

`x writer --aliasinit` does not work in Elvish shell. It seems you do support other non-posix shells.

https://github.com/x-cmd/x-cmd/blob/main/mod/writer/lib/alias

**Is is urgent**
No, low priority. 

**Additional context**

**Comment by iandol** [comment]
Maybe you consider this a duplicate of the more general: https://github.com/x-cmd/x-cmd/issues/21 -- maybe we can make a todo list there and close this?

**Comment by edwinjhlee** [comment]
Yes. Currently we don't provide writer aliases for nushell/evlish ...  
We will add these aliases in the next version.

**Comment by edwinjhlee** [comment]
This issue is supposed to resolved in v0.3.14 

Feel free to  confirm and close it.

**Comment by iandol** [comment]
I see you have added the aliases into `a` module, e.g. `fn en { |@a| use x; x:x writer --sendalias en $@a ; }` -- this is ok we can use `a:en` for example, but the docs for writer are not updated for this, perhaps you can add a comment for non-posix users?


Offtopic for this issue but I'll mention it here: I want to use LM Studio to be used by writer, but if I use `x writer --sendalias zh "Some words"` then it defaults to OpenAI. I tried `x writer tran --provider=lms --sendalias="zh" "Hello how are you?"` or others but no luck...

**Comment by edwinjhlee** [comment]
> I see you have added the aliases into `a` module, e.g. `fn en { |@a| use x; x:x writer --sendalias en $@a ; }` -- this is ok we can use `a:en` for example, but the docs for writer are not updated for this, perhaps you can add a comment for non-posix users?
> 
> Offtopic for this issue but I'll mention it here: I want to use LM Studio to be used by writer, but if I use `x writer --sendalias zh "Some words"` then it defaults to OpenAI. I tried `x writer tran --provider=lms --sendalias="zh" "Hello how are you?"` or others but no luck...



The `a:<agent>` design is only for elvish for elvish doesn't support alias staring with '@' . I will mention this diff in the https://www.x-cmd.com/start/elvish

---

About lms module, we only use it during development. Have you 'x lms init' ? 


---

Actually, by design, we will loop up AI service in the order: llmf/ollama/lms , gemini, openai, mistral , ...

So if you setup lms or ollama, the writer will first call ollama/lms .

---

However, we don't have enough time to use and check it. I personally use `ollama` and `@o` a lot. So I don't know the situation of lms. I will try using it later.

**Comment by edwinjhlee** [comment]
This is not supported.
```bash
x writer tran --provider=lms --sendalias="zh" "Hello how are you?"
```

This is supported.

```bash
x writer tran --provider lms --sendalias "zh" "Hello how are you?"
```

We might support --<key>=<value> in the future. But currently we don't.

**Comment by iandol** [comment]
I don't mind to use `--key value` format. This is not working:

```
x writer tran --provider lms --sendalias "zh" "Hello how are you?"

    LLO, WHAT'S GOING ON WITH YOU?
```

The server log shows:

```
[2024-07-11 20:41:48.540] [INFO] Received POST request to /v1/chat/completions with body: {
  "model": "",
  "messages": [
    {
      "role": "system",
      "content": "\nYou are a translate assistant to help the following text to 'English' .\nNOTICE: Your response MUST ONLY be the text translated. NO extra.\n\nThe text to be translated is as below:\n"
    },
    {
      "role": "user",
      "content": "--sendalias zh Hello how are you?"
    }
  ],
  "seed": 0,
  "temperature": 0.7,
  "stream": true
}
```

Note --sendalias zh is not being parsed.

**Comment by iandol** [comment]
If I swap parameters it is worse: `x writer tran --sendalias "zh" --provider "lms" "Hello how are you?"`

```
[2024-07-11 20:45:42.902] [INFO] Received POST request to /v1/chat/completions with body: {
  "model": "gpt-4o",
  "messages": [
    {
      "role": "system",
      "content": "\nYou are a translate assistant to help the following text to 'English' .\nNOTICE: Your response MUST ONLY be the text translated. NO extra.\n\nThe text to be translated is as below:\n"
    },
    {
      "role": "user",
      "content": "--sendalias zh --provider lms Hello how are you?"
    }
  ],
  "max_tokens": 1000,
  "seed": 0,
  "stream": true
}
```

**Comment by edwinjhlee** [comment]
> If I swap parameters it is worse: `x writer tran --sendalias "zh" --provider "lms" "Hello how are you?"` -- note it always passes 'English' in the system message.
> 
> ```
> [2024-07-11 20:45:42.902] [INFO] Received POST request to /v1/chat/completions with body: {
>   "model": "gpt-4o",
>   "messages": [
>     {
>       "role": "system",
>       "content": "\nYou are a translate assistant to help the following text to 'English' .\nNOTICE: Your response MUST ONLY be the text translated. NO extra.\n\nThe text to be translated is as below:\n"
>     },
>     {
>       "role": "user",
>       "content": "--sendalias zh --provider lms Hello how are you?"
>     }
>   ],
>   "max_tokens": 1000,
>   "seed": 0,
>   "stream": true
> }
> ```
> 
> --sendalias is not mentioned on the help page: https://www.x-cmd.com/mod/writer/tran -- but it is useful!


sendalias is design to setup aliases starting with '@' . 
It is supposed to be the inner function. We are going to expose it using a better/short name.

To be honest, I don't use the '@en' and '@zh'. That's the reason there is a lot of problem in it.

It takes us serveral days to deal with it. We are busy redesigning and working on hub module.

**Comment by edwinjhlee** [comment]
This issue was resolved in v0.3.14 ~
Please reopen this issue if the problem persists.

Thanks for your support ~
